### PR TITLE
UX improvements of DAG tag filtering

### PR DESCRIPTION
**Pull request by ryanahamilton**
- Instead of requiring the user to select/deselect tags and then submit the form with the "Filter tags" button, the page now updates as the tag is selected/deselected.
- ~Removes the storage of the tags in the `tags_filter` cookie. For a filtering pattern such as this, the URL params should always reflect the current result set. Currently, you can have your result set filtered by tags and have no `tags=` params in your URL. This also required using a `&reset_tags=Reset` param to clear the values—this will no longer be needed with the URL explicitly reflecting the values.~ Updated the server-side logic, so that if cookie values exist, the `tags=` param is added to the URL so it matches the filtered result set. I also made an update to strip the `reset_tags=reset` from the URL after the cookie value has been cleared.
- Replaces the "Reset" button with an "X" icon button (same pattern used for search filter, added in #11583). This button only appears when tag filters are applied.
- The sum of these updates yields a much simpler UI where we don't need persistent buttons taking up space.
- Updates Docs screenshot

| Before | After |
|---|---|
|  Screen Recording 2020-10-19 at 10 29 28 AM | Screen Recording 2020-10-19 at 10 13 27 AM  |

CC: @zacharya19 thought you might be interested in these modifications to your feature originally added in #6489.

### CODE REVIEW

**Comment by ashb** [comment]
I think the reason for the `tags_filter` cookie is so that if you navigate away (say go and view a DAG) and then come back to the home page we want the same filters to apply.

**Comment by ryanahamilton** [comment]
@ashb yeah, I understand that desire, but I believe it to be anti-pattern for filtering to not have continuity with the URL params. When a user utilizes the back button, their previously applied tag filters will still persist.

**Comment by ashb** [comment]
From a user POV the existing behaviour is very useful though -- espeically if you have a large Airflow deployment, and you only want to see "your team's" DAGs -- having to re-apply the filter each time you visit the page would be tedious.

**Comment by ryanahamilton** [comment]
@ash I see that even though "tags" are typically an unspecified feature use, this feature was originally intended for this "team" use-case, which makes more sense to persist the filter. I'll see about refactoring this to keep the cookie in place.

**Comment by mik-laj** [comment]
I can see the benefits of @ryanahamilton 's solution, but also understand the needs of some users presented by @ashb  who wanted selected tags to be remembered at all times. Maybe we can combine these two ideas and let the user keep the currently selected tags as default? I see it this way, that the user can browse DAGs normally according to @ryanahamilton 's suggestion, but if user wants to configure, be able to quickly return to the selected configuration, all users has to do is press the "Save as default" button. It doesn't even have to be on this page, but it could be a configuration option in your user profile settings ("Default selected tags" or something similar). I think this will better address the needs of both user groups.

**Comment by ryanahamilton** [comment]
@mik-laj I like that idea. It might make sense to save the status (all/active/paused) in the "default" preference as well since we're also cookie-ing that value? That feels like it should be addressed as a feature in a subsequent PR—I think I can offer a refactor to this PR that maintains the current user expectations but still delivers on the UX improvements.

**Comment by mik-laj** [comment]
@ryanahamilton  I have no preference. Each solution works for me. Let me know when your change is ready for review.

File: airflow/www/views.py [hunk 1]
```diff
@@ -418,12 +418,15 @@ def get_int_arg(value, default=0):
         if request.args.get('reset_tags') is not None:
             flask_session[FILTER_TAGS_COOKIE] = None
             arg_tags_filter = None
```
Comment (zacharya19) [inline airflow/www/views.py]: No longer needed.

File: airflow/www/views.py [hunk 2]
```diff
@@ -418,12 +418,15 @@ def get_int_arg(value, default=0):
...
+            # Remove the reset_tags=reset from the URL
+            return redirect(url_for('Airflow.index'))
         else:
```
Comment (zacharya19) [inline airflow/www/views.py]: Since you're returning in the if block, you can remove the else condition here.

**Review by zacharya19 — COMMENTED** [review]
LGTM, a few nits, but I really like the improvements!

File: airflow/www/views.py (hunk 2 above)
Comment (ryanahamilton) [reply airflow/www/views.py]: Ah, yes, thanks for edits!

**Review by ashb — APPROVED** [review]
Nice approach

**Pull request merged**